server/services/star.js

```diff
@@ -438,15 +470,26 @@ module.exports = class StarService extends EventEmitter {
     }
 
     isDeadStar(star) {
-        return star.naturalResources <= 0;
+        return star.naturalResources.economy <= 0 && star.naturalResources.industry <= 0 && star.naturalResources.science <= 0;
     }
 
     deductNaturalResources(star, amount) {
-        star.naturalResources -= amount;
+        star.naturalResources.economy -= amount;
+        star.naturalResources.industry -= amount;
+        star.naturalResources.science -= amount;
 
+        if (Math.floor(star.naturalResources.economy) <= 0) {
+            star.naturalResources.economy = 0;
+        }
+        if (Math.floor(star.naturalResources.industry) <= 0) {
+            star.naturalResources.industry = 0;
+        }
+        if (Math.floor(star.naturalResources.science) <= 0) {
+            star.naturalResources.science = 0;
+        }
+        
         // if the star reaches 0 resources then reduce the star to a dead hunk.
-        if (star.naturalResources <= 0) {
-            star.naturalResources = 0;
+        if(star.naturalResources.economy <= 0 && star.naturalResources.industry <= 0 && star.naturalResources.science <= 0) {
```

Comment:
`this.isDeadStar`?

Reply:
Sure but I mean it didnt use that to start with :grin:

Reply:
But resolved anyway :wink: